wasm/derive/src/lib.rs

```diff
@@ -30,7 +30,7 @@ pub fn iroha_wasm(_: TokenStream, item: TokenStream) -> TokenStream {
 
     quote! {
         #[no_mangle]
-        unsafe extern "C" fn _iroha_wasm_main(ptr: u32, len: u32) {
+        unsafe extern "C" fn _iroha_wasm_main(ptr: usize, len: usize) {
```

Comment:
This is OK, because we have a fixed WASM32 target, and this size is known ahead of time. I would, however add a few provisional checks just in case `usize` ends up being `u64`.